lua/nvim-lsp-ts-utils/inlay-hints.lua

```diff
@@ -0,0 +1,91 @@
+local o = require("nvim-lsp-ts-utils.options")
+
+local M = {}
+
+M.state = {
+    enabled = false,
+    ns = vim.api.nvim_create_namespace("ts-inlay-hints"),
+}
+
+function M.setup_autocommands()
+    vim.cmd([[
+       augroup TSInlayHints
+       au BufEnter,BufWinEnter,TabEnter,BufWritePost *.ts,*.js,*.tsx,*.jsx :lua require'nvim-lsp-ts-utils'.autocmd_fun()
```

Comment:
We might want to play around with these autocommands, since currently changing text doesn't show / update hints until the user writes the file. Do we have any precedent from other implementations? `TextChanged` and `InsertLeave` seem to make sense but could be too aggressive.

Reply:
rust-tools uses these, because I think it didn't matter with rust analyzer. Might be worth investigating to update them while typing for a signature hint-esque flow

Reply:
Yup, it updates while typing. I think TextChanged and TextChangedI should be fine